core/src/main/java/bio/terra/pearl/core/service/search/terms/SearchTermParser.java

```diff
@@ -16,14 +16,26 @@ public T parseVariable(String variable) {
 
         List<String> splitModelName = List.of(variableNoBraces.split("\\.", 2));
 
-        if (!splitModelName.getFirst().equals(this.getTermName())) {
+        String modelName = splitModelName.get(0);
+        String studyName = null;
+
+        if (modelName.contains("[")) {
```

Comment:
this feels like parser logic that belongs more in the antlr parser.  Is it possible to have a VARIABLE_WITH_STUDY term?  or otherwise update our parsing so that the syntax for how studies are specified is handled there?

Reply:
I have ideas on how to do this but my antlr skills are not nearly refined enough to accomplish them. I tried doing
```antlr
expr:  term OPERATOR term | expr AND expr | expr OR expr | NOT expr | INCLUDE PAR_OPEN term PAR_CLOSE | PAR_OPEN expr PAR_CLOSE;
term: value | function | variable;
value: NUMBER | STRING  | BOOLEAN | NULL;
function: FUNCTION_NAME PAR_OPEN term (',' term)* PAR_CLOSE;
variable: VAR_OPEN VAR_MODEL (VAR_STUDY)? (VAR_ARGUMENTS)? VAR_CLOSE;

// Lexer rules
NUMBER: [0-9]+ ('.' [0-9]+)?;
STRING: '\'' (~[\\'\r\n])* '\'';
VAR_STUDY: '["'[a-zA-Z0-9_]+'"]';
VAR_MODEL: [a-zA-Z0-9_]+;
VAR_ARGUMENTS: '.'([a-zA-Z0-9_]|'.')+;
VAR_SEPARATOR: '.';
VAR_OPEN: '{';
VAR_CLOSE: '}';
```
But there's some weird execution order issues where it's trying to read functions as variables even though there isn't a {} surrounding them. I can't figure out why... I might punt this off as a future improvement.